shared/common-adapters/header-hoc.native.js

```diff
@@ -4,54 +4,127 @@ import Text from './text'
+  floatingMenuVisible: boolean,
+|}
+export class HeaderHocHeader extends React.Component<Props, State> {
+  state = {
+    floatingMenuVisible: false,
+  }
+  _onHidden = () => this.setState({floatingMenuVisible: false})
+  _showFloatingMenu = () => this.setState({floatingMenuVisible: true})
+  render() {
+    // TODO: remove these after updates are fully integrated
+    const onLeftAction = this.props.onLeftAction || this.props.onBack || this.props.onCancel
+    const leftAction = this.props.leftAction || this.props.onCancel ? 'cancel' : this.props.onBack ? 'back' : null
+    const rightActions = this.props.rightActions || (this.props.onRightAction && this.props.rightActionLabel) ? [{
+      label: this.props.rightActionLabel,
+      onPress: this.props.onRightAction,
+    }] : null
+
+    return (
+      <Box
+        style={Styles.collapseStyles([
+          styles.header,
+          this.props.borderless && styles.borderless,
+          this.props.headerStyle,
+        ])}
+      >
+        {this.props.customComponent}
+        <Box style={Styles.collapseStyles([styles.leftAction, this.props.titleComponent && styles.unflex])}>
+          {onLeftAction &&
+            (leftAction === 'cancel' ? (
+              <Text type="BodyBigLink" style={styles.action} onClick={onLeftAction}>
+                {this.props.leftActionText || this.props.customCancelText || 'Cancel'}
+              </Text>
+            ) : (
+              <BackButton
+                badgeNumber={this.props.badgeNumber}
+                hideBackLabel={this.props.hideBackLabel}
+                iconColor={
+                  this.props.theme === 'dark' ? Styles.globalColors.white : Styles.globalColors.black_40
+                }
+                style={styles.action}
+                onClick={onLeftAction}
+              />
+            ))}
+        </Box>
+        <Box
+          style={Styles.collapseStyles([
+            styles.titleContainer,
+            !this.props.rightActions && styles.titlePadding,
+          ])}
         >
-          {rightActionLabel}
-        </Text>
+          {!!this.props.title && !this.props.titleComponent
+            ? (<Text type="BodySemibold" style={styles.title} lineClamp={1}>
+                {this.props.title}
+              </Text>)
+            : (this.props.titleComponent)
+          }
+        </Box>
+        <Box style={Styles.collapseStyles([styles.rightActions, this.props.titleComponent && styles.unflex])}>
+          <Box style={styles.rightActionsWrapper}>
+            {rightActions &&
+              rightActions
+                .filter(Boolean)
+                .slice(
+                  0,
+                  rightActions && rightActions.length <= MAX_RIGHT_ACTIONS
+                    ? MAX_RIGHT_ACTIONS
+                    : MAX_RIGHT_ACTIONS - 1
+                )
+                .map((action, item) => renderAction(action))}
+            {rightActions &&
+              rightActions.length > MAX_RIGHT_ACTIONS && (
```

Comment:
you probably want to do the filtering before doing the length check, i.e.

```
const rightActions = this.props.rightActions ? this.props.rightActions.filter(Boolean) : ...
```

Reply:
Good point!